xds/internal/balancer/xdsrouting/matcher_path.go

```diff
@@ -0,0 +1,99 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xdsrouting
+
+import (
+	"regexp"
+	"strings"
+
+	"google.golang.org/grpc/balancer"
+)
+
```

Comment:
I think it might be better to define separate interfaces for `pathMathers` and `headerMatchers` for a couple of reasons.
1. Path matching only requires the method name and none of the other fields from the `balancer.PickInfo`.
2. Path matchers and header matchers are probably not going to be used interchangeably in the LB policy.

Reply:
Done